src/drivers/wayland/wayland.c

```diff
@@ -0,0 +1,2555 @@
+        /* All subsequent buffers damage is recorded, initialize by applying
+         * their damage to this buffer
+         */
+        for (src = smm_next_buffer(buf); src != NULL; src = smm_next_buffer(src)) {
+            src_base = smm_map(src);
+            if (src_base == NULL) {
+                LV_LOG_ERROR("cannot map source buffer to copy from");
+                goto done;
+            }
+
+            cache_apply_areas(obj->window, buf_base, src_base, src);
+        }
+
+        /* Purge out-of-date cached damage (up to and including next buffer) */
+        src = smm_next_buffer(buf);
+        if (src == NULL) {
+            cache_purge(obj->window, src);
+        }
+    }
+
+    fail_init = false;
+done:
+    return fail_init;
+}
+
+static void sme_free_buffer(void* ctx, smm_buffer_t* buf)
+{
+    struct wl_buffer* wl_buf = SMM_BUFFER_PROPERTIES(buf)->tag[LW_TAG_LOCAL];
+    wl_buffer_destroy(wl_buf);
+}
+
+static struct lw_surface_object* create_surface_obj(struct lw_application* app, lw_window_t* window,
+    enum lw_object_type type,
+    struct lw_surface_object* parent)
+{
+    struct lw_surface_object* obj = malloc(sizeof(*obj));
+    LV_ASSERT_MALLOC(obj);
+    if (!obj) {
+        goto err_out;
+    }
+
+    memset(obj, 0x00, sizeof(struct lw_surface_object));
+
+    obj->surface = wl_compositor_create_surface(app->compositor);
+    if (!obj->surface) {
+        LV_LOG_ERROR("cannot create surface for graphic object");
+        goto err_free;
+    }
+
+    obj->buffer_group = smm_create_group();
+    if (obj->buffer_group == NULL) {
+        LV_LOG_ERROR("cannot create buffer group for graphic object");
+        goto err_destroy_surface;
+    }
+
+    obj->window = window;
+    obj->type = type;
+    obj->surface_configured = true;
+    obj->pending_buffer = NULL;
+    wl_surface_set_user_data(obj->surface, obj);
+    SMM_TAG(obj->buffer_group, LW_TAG_LOCAL, obj);
+
+    return obj;
+
+err_destroy_surface:
+    wl_surface_destroy(obj->surface);
+
+err_free:
+    free(obj);
+err_out:
+
+    return NULL;
+}
+
+static void destroy_wl_surface_obj(struct lw_surface_object* obj)
+{
+    if (obj->subsurface) {
+        wl_subsurface_destroy(obj->subsurface);
+    }
+
+    wl_surface_destroy(obj->surface);
+    smm_destroy_group(obj->buffer_group);
+    free(obj);
+}
+
+static bool resize_window(lw_window_t* window, int width, int height)
+{
+    lv_color32_t* buf1 = NULL;
+    struct smm_buffer_t* body_buf1;
+    //struct smm_buffer_t* body_buf2;
+    int b;
+
+    LV_LOG_TRACE("resize window %dx%d frame: %ld rendered: %d",
+        width, height, window->frame_counter, window->frame_done);
+
+/* Update size for newly allocated buffers */
+    smm_resize_group(window->body->buffer_group, ((width * LW_COLOR_PX_SIZE) * height) * 2);
+
+    window->width = width;
+    window->height = height;
+
+    window->body->width = width;
+    window->body->height = height;
+
+    /* Pre-allocate two buffers for the window body here */
```

Comment:
Unfortunately, commenting this out will break compatibility with weston

Reply:
I just compiled LVGL with this driver on debian (gnome 46.2/ mutter46.2) and ran it. with LVGL widgets  animations window activated, demo application showed 60~100+ FPS, lower than 20%CPU used (intel 11th i7, really CPU used lowerthan 5%), it worked fine. 
I have not tested demo with weston, so there would have problems..

Reply:
I just compiled LVGL with this driver on debian (gnome 46.2/ mutter46.2) and ran it. with LVGL widgets  animations window activated, demo application showed 60~100+ FPS, lower than 20%CPU used (intel 11th i7, really CPU used lowerthan 5%), it worked fine. 
I have not tested demo with weston, so there would have problems..